src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,116 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './TodoList.scss';
+import classNames from 'classnames';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
+type Props = {
+  todos: Todo[],
+  setSelectedUserId: (userId: number) => void,
+  selectedTodoId: number,
+  setSelectedTodoId: (todoId: number) => void,
+};
+
+function shuffleArr(arr: Todo[]) {
+  return arr.sort(() => Math.random() - 0.5);
+}
+
+export const TodoList: React.FC<Props> = React.memo(({
+  todos,
+  setSelectedUserId,
+  selectedTodoId,
+  setSelectedTodoId,
+}) => {
+  const [query, setQuery] = useState('');
+  const [selectValue, setSelectValue] = useState('all');
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+
+  useEffect(() => {
+    setVisibleTodos(todos.filter(todo => {
+      if (!todo.title.includes(query.toLowerCase())) {
```

Comment:
`query.toLowerCase()` - better to calculate it before filtering but not do it on every iteration